src/backend_connection/error.rs

```diff
@@ -1,15 +1,60 @@
 // Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 (eventually AGPL-3.0) 2016-present Scille SAS
 
-use pyo3::{exceptions::PyException, pyclass};
+use pyo3::{exceptions::PyException, PyErr};
+use reqwest::StatusCode;
 
-use libparsec::client_connection;
+pyo3::create_exception!(_parsec, BackendConnectionError, PyException);
+pyo3::create_exception!(_parsec, BackendProtocolError, BackendConnectionError);
+pyo3::create_exception!(_parsec, BackendNotAvailable, BackendConnectionError);
+pyo3::create_exception!(_parsec, BackendConnectionRefused, BackendConnectionError);
+pyo3::create_exception!(
+    _parsec,
+    BackendInvitationAlreadyUsed,
+    BackendConnectionRefused
+);
+pyo3::create_exception!(_parsec, BackendInvitationNotFound, BackendConnectionRefused);
+// TODO: hack needed by `LoggedCore.get_user_info`
```

Comment:
Which hack needed ?

Reply:
It's the comment from the python code

Reply:
That don't answer the question sadly

Reply:
I don't even know, I migrate the code only

Reply:
@touilleMan Do you have any info ? 
this comment is missing information to be relevant

Reply:
This error is still needed: 
https://github.com/Scille/parsec-cloud/blob/7b9c1267e666a954cb5e9c55319700f03fe2019a/parsec/core/logged_core.py#L191-L193

The idea it was convenient to use `BackendError` in `LoggedCore.get_user_info` instead of exposing our own exceptions. Of course we don't want to do that in the Rust codebase (see the high level api where we define a different error for each api function :smile: ), but it's okay to keep this code in the Python codebase

Feel free to improve the comment if you think `LoggedCore.get_user_info` is not precise enough to point the reader where this error is used ;-)

Reply:
```suggestion
// TODO: `LoggedCore.get_user_info` raise `BackendNotFoundError`
// It was convenient to have it inherit `BackendConnectionError` but it would be best to have a specific set of `exception` for it. 
```